connexion/app.py

```diff
@@ -134,7 +134,12 @@ def add_api(self, swagger_file, base_path=None, arguments=None, auth_all_paths=N
         # TODO test if base_url starts with an / (if not none)
         arguments = arguments or dict()
         arguments = dict(self.arguments, **arguments)  # copy global arguments and update with api specfic
-        yaml_path = self.specification_dir / swagger_file
+
+        if isinstance(swagger_file, dict):
+            yaml_path = swagger_file
+        else:
+            yaml_path = self.specification_dir / swagger_file
+
         api = Api(swagger_yaml_path=yaml_path,
```

Comment:
Would be better to rename this parameter to fit in a more descriptive way what this argument is expected to be. In this case it can be a yaml path is clear to be passed here after reading `swagger_yaml_path` but this being able to be a `dict` is very misleading.